### PR TITLE
[MM-20684] Slash command autocomplete

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR will add autocomplete_suggestions endpoint doc

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-21493

### CODE REVIEW

File: v4/source/commands.yaml [hunk 1]
```diff
@@ -166,6 +166,54 @@
...
             listCommands, resp := Client.ListAutocompleteCommands(<TEAMID>)
+  '/teams/{team_id}/commands/autocomplete_suggestions':
+    get:
+      tags:
+        - commands
+      summary: List commands' autocomplete data
+      description: |
+        List commands' autocomplete data for the team.
+        ##### Permissions
+        `view_team` for the team.
```
Comment (hanzei) [inline v4/source/commands.yaml:177]: Please add the minimum server version similar to https://github.com/mattermost/mattermost-api-reference/blob/7287dc4fc9b556e7cec04f6753f5c3d7faba7f3d/v4/source/bots.yaml#L10

File: v4/source/commands.yaml [hunk 2]
```diff
@@ -166,6 +166,54 @@
...
+      parameters:
+        - name: team_id
+          in: path
+          description: Team GUID
+          required: true
+          schema:
+            type: string
+        - name: user_input
+          in: query
+          description: String inputted by the user.
```
Comment (hanzei) [inline v4/source/commands.yaml:187]: Is this really an optional parameter?

File: v4/source/commands.yaml [hunk 3]
```diff
@@ -166,6 +166,54 @@
...
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/AutocompleteSuggestion"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
```
Comment (hanzei) [inline v4/source/commands.yaml:204]: Out of curiosity: When does this happen?

File: v4/source/definitions.yaml [hunk 4]
```diff
@@ -786,6 +786,24 @@ components:
...
+          description: Predicted text user might want to input
+          type: string
+        Hint:
+          description: Hint about suggested input
+          type: string
+        Description: 
+          description: Description of the suggested command
+          type: string
+        IconData:
+          description: Icon data in base64 format
```
Comment (hanzei) [inline v4/source/definitions.yaml:805]: The documentation should make clear that this is an scg image.

Comment (levb) [reply v4/source/definitions.yaml:805]: see my comment in the source about maybe renaming to be more intention-revealing

File: v4/source/commands.yaml (hunk 3 above)
Comment (iomodo) [reply v4/source/commands.yaml:204]: when you want to get commands for a team and you do not have a permission(for that team).

File: v4/source/commands.yaml [hunk 5]
```diff
@@ -166,6 +166,58 @@
...
+    get:
+      tags:
+        - commands
+      summary: List commands' autocomplete data
+      description: |
+        List commands' autocomplete data for the team.
+        ##### Permissions
+        `view_team` for the team.
+
+
```
Comment (hanzei) [inline v4/source/commands.yaml:179]: Nit: Remove this line for consistency
```suggestion

```

File: v4/source/commands.yaml (hunk 3 above)
Comment (hanzei) [reply v4/source/commands.yaml:208]: Thanks for the explanation :+1:

**Comment by levb** [comment]
I'll leave to @iomodo or @hanzei to merge when appropriate.

**Pull request merged**